### PR TITLE
Check on mix new if the given application module exists and raises if so

### PR DESCRIPTION
**Pull request by lpil**
My attempt to do #3275. What are your thoughts? :)

Also, is there a way to run just one file or module or tests in this project? I couldn't work out how to do anything more specific than `make test`.

Cheers,
Louis

### CODE REVIEW

File: lib/mix/lib/mix/tasks/new.ex [hunk 1]
```diff
@@ -157,12 +158,19 @@ defmodule Mix.Tasks.New do
...
 
-  defp check_mod_name!(name) do
+  defp check_mod_name_validity!(name) do
     unless name =~ ~r/^[A-Z]\w*(\.[A-Z]\w*)*$/ do
       Mix.raise "Module name must be a valid Elixir alias (for example: Foo.Bar), got: #{inspect name}"
     end
   end
 
+  defp check_mod_name_availability!(name) do
+    name = String.to_atom("Elixir.#{name}")
```
Comment (josevalim) [inline lib/mix/lib/mix/tasks/new.ex]: `Module.concat(Elixir, name)`

**Comment by josevalim** [comment]
It looks great! I have just added a tiny comment. We would love if you could send a similar pull request to `mix phoenix.new` in the phoenix project (check the installer folder).

File: lib/mix/lib/mix/tasks/new.ex (hunk 1 above)
Comment (lpil) [inline lib/mix/lib/mix/tasks/new.ex]: Ah, fantastic. Thank you.

**Comment by lpil** [comment]
How's that? :)

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by lpil** [comment]
Excellent. Thanks. Looking forward to more `Level:Starter`'s ;)

**Pull request merged**